src/v/cluster/archival_metadata_stm.cc

```diff
@@ -696,18 +689,16 @@ archival_metadata_stm::sync(model::timeout_clock::duration timeout) {
     // below. If replication failed, exit unsuccessfully.
 
     if (_last_replicate) {
-        auto fut = std::exchange(_last_replicate, std::nullopt).value();
-
-        if (!fut.available()) {
+        if (!_last_replicate->result.available()) {
```

Comment:
>Use a shared future for _last_replicate so that it can be awaited
>multiple times and reset it only after it is resolved.

Just trying to understand the remaining problematic scenarios. Does this generalize to: `sync()` may be called concurrently by different fibers?

Reply:
Yes, it does. To give one example at least, ListOffsets kafka rpc causes a sync() call too.